Kernel/FileSystem/FATFS/FileSystem.cpp

```diff
@@ -10,6 +10,77 @@
 
 namespace Kernel {
 
+DOSBIOSParameterBlockVersion DOSBIOSParameterBlock::get_bpb_version() const
+{
+    // A DOS 7 EBPB should _never_ contain the values 0x28 or 0x29 at
+    // the offset associated with `dos4_block->signature`
+    // (aka `dos7_block->sectors_per_fat_32bit`) due to the maximum number of
+    // clusters ensuring the number of sectors per fat will not exceed 0x200000.
+    // As a result, it should be safe to determine BPB version through the
+    // signature fields by checking the DOS 4 signature offset prior to the DOS 7 one.
+    if (dos4_block->signature == 0x28) {
+        return DOS_BPB_3;
+    } else if (dos4_block->signature == 0x29) {
+        return DOS_BPB_4;
+    } else if (dos7_block->signature == 0x28 || dos7_block->signature == 0x29) {
+        return DOS_BPB_7;
+    } else {
+        return DOS_BPB_UNKNOWN;
+    }
+}
+
+DOS4BIOSParameterBlock const* DOSBIOSParameterBlock::get_dos4_bpb() const
+{
+    // Only return if signature indicates this portion of the block is filled out.
```

Comment:
We always return from the method :). "Only return *parameter block* if the signature ...." ?